### PR TITLE
Add `constexpr` to static member functions for point types, add macro for `if constexpr`

### PR DESCRIPTION
**Pull request by al-tu**
a small subset of related [to constexpr point constructors MR] changes

### CODE REVIEW

File: common/include/pcl/pcl_macros.h [hunk 1]
```diff
@@ -438,3 +438,9 @@ aligned_free (void* ptr)
 #else
   #define PCL_NODISCARD
 #endif
+
+#if (__cplusplus >= 201703L) || (defined(_MSC_VER) && (_MSC_VER >= 1910) && (_MSVC_LANG >= 201703L))
```
Comment (kunaltyagi) [inline common/include/pcl/pcl_macros.h:442]: What about using std feature test macro `__cpp_if_constexpr`?

File: sample_consensus/include/pcl/sample_consensus/impl/sac_model_registration.hpp [hunk 2]
```diff
@@ -55,9 +55,12 @@ pcl::SampleConsensusModelRegistration<PointT>::isSampleGood (const Indices &samp
...
 
-  PointT p10 = (*input_)[samples[1]] - (*input_)[samples[0]];
-  PointT p20 = (*input_)[samples[2]] - (*input_)[samples[0]];
-  PointT p21 = (*input_)[samples[2]] - (*input_)[samples[1]];
+  PointT p10;
+  pcl::copyPoint((*input_)[samples[1]] - (*input_)[samples[0]], p10);
+  PointT p20;
+  pcl::copyPoint((*input_)[samples[2]] - (*input_)[samples[0]], p20);
+  PointT p21;
+  pcl::copyPoint((*input_)[samples[2]] - (*input_)[samples[1]], p21);
```
Comment (kunaltyagi) [inline sample_consensus/include/pcl/sample_consensus/impl/sac_model_registration.hpp:63]: Unrelated?

Comment (mvieth) [reply sample_consensus/include/pcl/sample_consensus/impl/sac_model_registration.hpp:63]: I believe this is to fix an error while compiling with PointXYZLAB, I noticed that too while compiling yesterday. But I think a better fix would be to use the Eigen mappings `Eigen::Vector3f p10 = (*input_)[samples[1]].getVector3fMap() - (*input_)[samples[0]].getVector3fMap();` and/or to add an additional [constructor like the other point types have](https://github.com/PointCloudLibrary/pcl/blob/master/common/include/pcl/impl/point_types.hpp#L749). Also we should do that in a separate pull request to get that merged as soon as possible.

Comment (kunaltyagi) [reply sample_consensus/include/pcl/sample_consensus/impl/sac_model_registration.hpp:63]: >  Also we should do that in a separate pull request to get that merged as soon as possible.

Seconded

Comment (al-tu) [reply sample_consensus/include/pcl/sample_consensus/impl/sac_model_registration.hpp]: reverted

**Review by kunaltyagi — APPROVED** [review]

**Comment by kunaltyagi** [comment]
Successfully compiled this PR with C++17 flag, but didn't test if the constexpr functions actually were compile time only

**Review by mvieth — APPROVED** [review]

**Review by larshg — APPROVED** [review]

**Comment by al-tu** [comment]
> Successfully compiled this PR with C++17 flag, but didn't test if the constexpr functions actually were compile time only

https://github.com/PointCloudLibrary/pcl/pull/4752 follow up for this MR, covered with static_asserts. didn't cover c++17 macro to be expanded to constexpr if
